lib/cryptoexchange/exchanges/trade_satoshi/services/market.rb

```diff
@@ -0,0 +1,65 @@
+module Cryptoexchange::Exchanges
+  module TradeSatoshi
+    module Services
+      class Market < Cryptoexchange::Services::Market
+        class << self
+          def supports_individual_ticker_query?
+            false
+          end
+        end
+
+        def fetch
+          output = super ticker_url
+          adapt_all(output)
+        end
+
+        def ticker_url
+          "#{Cryptoexchange::Exchanges::TradeSatoshi::Market::API_URL}/getmarketsummaries"
+        end
+
+        def adapt_all(output)
+          output['result'].map do |ticker|
+            base, target = ticker['market'].split('_')
+            market_pair = Cryptoexchange::Models::MarketPair.new(
+                            base: base,
+                            target: target,
+                            market: TradeSatoshi::Market::NAME
+                          )
+            adapt(ticker, market_pair)
+          end
+        end
+
+        # {
+        #   "market":"PAK_BTC",
+        #   "high":0.00000000,
+        #   "low":0.00000000,
+        #   "volume":0.00000000,
+        #   "baseVolume":0.00000000,
+        #   "last":0.00000070,
+        #   "bid":0.00000075,
+        #   "ask":0.00000514,
+        #   "openBuyOrders":41,
+        #   "openSellOrders":13,
+        #   "change":0.0
+        # },
+        def adapt(output, market_pair)
+          market = output
+          # raise output.inspect
+          ticker = Cryptoexchange::Models::Ticker.new
+          ticker.base = market_pair.base
+          ticker.target = market_pair.target
+          ticker.market = TradeSatoshi::Market::NAME
+          ticker.ask = NumericHelper.to_d(market['sell'])
+          ticker.bid = NumericHelper.to_d(market['buy'])
+          ticker.last = NumericHelper.to_d(market['last'])
+          ticker.high = NumericHelper.to_d(market['high'])
+          ticker.low = NumericHelper.to_d(market['low'])
+          ticker.volume = NumericHelper.to_d(market['baseVolume'])
```

Comment:
and according to respond:
```
<Cryptoexchange::Models::Ticker:0x00007f7fb0b1ff80 @base="BOLI", @target="BTC", @market="trade_satoshi", @ask=nil, @bid=nil, @last=0.65e-5, @high=0.75e-5, @low=0.65e-5, @volume=0.610584e-2, @timestamp=1515724811, @payload={"market"=>"BOLI_BTC", "high"=>7.5e-06, "low"=>6.5e-06, "volume"=>898.86401338, "baseVolume"=>0.00610584, "last"=>6.5e-06, "bid"=>6.5e-06, "ask"=>7.64e-06, "openBuyOrders"=>37, "openSellOrders"=>89, "change"=>0.0}
```
see in payload, maybe volume needs to be `volume` because it is `base` coin's volume

Reply:
I wasn't sure if volume was to be based on BTC or on the base currency.  The Implementations guide should be improved to state what's expected.  It says to make sure we get it right as "Some exchanges may not follow convention and get this mixed up" but examples given in points 1), 2) and 3) does not really clear it up since BTC, USD, and ETH variously appears on both sides depending on exchange.

Reply:
baseVolume = volume expressed in BTC's
volume = volume expressed in whatever is the "base" of the Cryptoexchange::Pair